server/store/sqlstore/store.go

```diff
@@ -1007,10 +1012,21 @@ func (s *SQLStore) GetSizeOfWhitelist() (int, error) {
 	return result, nil
 }
 
-func (s *SQLStore) StoreUserInWhitelist(userID string) error {
+func (s *SQLStore) StoreUserInWhitelist(tx *sql.Tx, userID string) error {
 	query := s.getQueryBuilder().Insert(whitelistedUsersTableName).Columns("mmUserID").Values(userID)
-	_, err := query.Exec()
+	var err error
+	if tx == nil {
+		_, err = query.Exec()
+	} else {
+		_, err = query.RunWith(tx).Exec()
+	}
 	if err != nil {
+		// We are checking two different strings here as the error is different for MySQL and Postgres
+		if strings.Contains(err.Error(), "Duplicate entry") || strings.Contains(err.Error(), "duplicate key value") {
+			s.api.LogDebug("UserID already present in whitelist", "UserID", userID)
+			return nil
+		}
```

Comment:
Let's copy/paste the more robust implementation from the server: https://github.com/mattermost/mattermost/blob/e5adf25fa095e6495d730233d39a45641d01b250/server/channels/store/sqlstore/store.go#L1157-L1174